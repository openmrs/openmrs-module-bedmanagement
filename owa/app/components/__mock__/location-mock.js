Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});