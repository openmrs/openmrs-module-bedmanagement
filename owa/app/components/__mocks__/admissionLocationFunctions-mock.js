import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
