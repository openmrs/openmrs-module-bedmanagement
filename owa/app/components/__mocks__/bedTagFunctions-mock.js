import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
