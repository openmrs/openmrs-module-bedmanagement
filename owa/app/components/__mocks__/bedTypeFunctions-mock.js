import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
