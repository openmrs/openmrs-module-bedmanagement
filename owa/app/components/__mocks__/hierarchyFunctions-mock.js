import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;