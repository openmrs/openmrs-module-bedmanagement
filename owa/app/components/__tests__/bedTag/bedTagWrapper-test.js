import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import BedTagWrapper from 'components/bedTag/bedTagWrapper';
require('components/__mocks__/location-mock');
require('babel-polyfill');
const testProps = {
    match:{
        isExact: true,
        params:{},
        path:'/owa/bedmanagement/bedTags.html',
        url:'/owa/bedmanagement/bedTags.html'
    },
    sleep: (milisec) => {
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                return resolve(true);
            }, milisec);
        });
    }
};

describe('BedTagWrapper', () => {
    beforeAll(() => {
        var mock = new MockAdapter(axios);
        const data = {
            results: [{
                'id': 2,
                'name': 'Broken',
                'uuid': 'e67ce6b1-8382-4514-80db-6d67a6611534'
            },
            {
                'id': 3,
                'name': 'Oxygen',
                'uuid': '25d9b2c8-7b70-48a5-a033-1051dcf7cc8a'
            },
            {
                'id': 4,
                'name': 'Isolation',
                'uuid': 'ff7ed494-7b9c-4478-812a-5187e297f94c'
            },
            {
                'id': 5,
                'name': 'Sanitation required',
                'uuid': '39376170-c020-4e19-9960-c6f9044886fd'
            }]
        };

        mock.onGet('https://192.168.33.10/openmrs/ws/rest/v1/bedTag')
            .reply(200, data);
    });

    it('Should render bed Tag page properly', async () =>{
        let bedTagWrapper = shallow(<BedTagWrapper match={testProps.match}/>);
        const bedTagFunctions = bedTagWrapper.instance().bedTagFunctions;

        await testProps.sleep(100);
        expect(bedTagWrapper.find('BedTagList').length).toBe(1);
        expect(shallowToJson(bedTagWrapper)).toMatchSnapshot();

        bedTagFunctions.setState({
            activePage: 'addEdit',
            pageData: {
                operation: 'add',
                bedTagId: null
            }
        });

        bedTagWrapper.update();
        expect(bedTagWrapper.find('AddEditBedTag').length).toBe(1);
        expect(shallowToJson(bedTagWrapper)).toMatchSnapshot();
    });

    it('Should work functions properly', async () =>{
        const bedTagWrapper = shallow(<BedTagWrapper match={testProps.match}/>);
        const bedTagFunctions = bedTagWrapper.instance().bedTagFunctions;

        expect(bedTagFunctions.getBedTags()).toEqual([]);

        await testProps.sleep(100);
        expect(bedTagFunctions.getBedTags()).toEqual([
            {
                'id': 2,
                'name': 'Broken',
                'uuid': 'e67ce6b1-8382-4514-80db-6d67a6611534'
            },
            {
                'id': 3,
                'name': 'Oxygen',
                'uuid': '25d9b2c8-7b70-48a5-a033-1051dcf7cc8a'
            },
            {
                'id': 4,
                'name': 'Isolation',
                'uuid': 'ff7ed494-7b9c-4478-812a-5187e297f94c'
            },
            {
                'id': 5,
                'name': 'Sanitation required',
                'uuid': '39376170-c020-4e19-9960-c6f9044886fd'
            }]);

        expect(bedTagFunctions.getBedTagByUuid('e67ce6b1-8382-4514-80db-6d67a6611534')).toEqual({
            'id': 2,
            'name': 'Broken',
            'uuid': 'e67ce6b1-8382-4514-80db-6d67a6611534'
        });

        expect(bedTagFunctions.getBedTagByUuid('ff7ed494-7b9c-4478-812a-5187e297f94c')).toEqual({
            'id': 4,
            'name': 'Isolation',
            'uuid': 'ff7ed494-7b9c-4478-812a-5187e297f94c'
        });


        bedTagFunctions.setState({
            activePage: 'addEdit',
            pageData: {
                operation: 'add',
                bedTagId: null
            }
        });

        expect(bedTagWrapper.state().activePage).toEqual('addEdit');
        expect(bedTagWrapper.state().pageData).toEqual({
            operation: 'add',
            bedTagId: null
        });
    });

});