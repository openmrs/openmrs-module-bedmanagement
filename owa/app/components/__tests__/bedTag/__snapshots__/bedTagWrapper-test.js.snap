// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedTagWrapper Should render bed Tag page properly 1`] = `
<div>
  <Notify />
  <Header
    path="/owa/bedmanagement/bedTags.html"
  />
  <div
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    <BedTagList
      bedTagFunctions={
        Object {
          "fetchBedTags": [Function],
          "getBedTagByUuid": [Function],
          "getBedTags": [Function],
          "notify": [Function],
          "setState": [Function],
        }
      }
      bedTags={Array []}
    />
  </div>
</div>
`;

exports[`BedTagWrapper Should render bed Tag page properly 2`] = `
<div>
  <Notify />
  <Header
    path="/owa/bedmanagement/bedTags.html"
  />
  <div
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    <AddEditBedTag
      bedTagFunctions={
        Object {
          "fetchBedTags": [Function],
          "getBedTagByUuid": [Function],
          "getBedTags": [Function],
          "notify": [Function],
          "setState": [Function],
        }
      }
      operation="add"
    />
  </div>
</div>
`;
